builder/site.hs

```diff
@@ -1,15 +1,39 @@
 --------------------------------------------------------------------------------
+{-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE OverloadedStrings #-}
-import           Data.Monoid           ((<>))
+import           Control.Monad
+import           Data.Aeson
+import qualified Data.ByteString.Lazy          as BL
+import           Data.Monoid                   ((<>))
 import           Data.Time.Calendar
 import           Data.Time.Clock
 import           Hakyll
+import           Hakyll.Core.Compiler
+import           Hakyll.Core.Compiler.Internal
```

Comment:
I'm feeling uneasy about importing internal modules. Is there really no way to get these from a supported interface?